vfs/dir.go

```diff
@@ -1085,12 +1085,26 @@ func (d *Dir) Mkdir(name string) (*Dir, error) {
 		return nil, err
 	}
 	// fs.Debugf(path, "Dir.Mkdir")
-	err = d.f.Mkdir(context.TODO(), path)
-	if err != nil {
-		fs.Errorf(d, "Dir.Mkdir failed to create directory: %v", err)
-		return nil, err
+	var fsDir *fs.Dir
```

Comment:
I think `operations.MkdirMetadata` does exactly this. I don't think it existed at the time you wrote this PR!